python/cudf/cudf/_lib/csv.pyx

```diff
@@ -0,0 +1,338 @@
+    )
+    XZ = (
+        <underlying_type_t_compression> compression_type.XZ
+    )
+
+
+cdef read_csv_args make_read_csv_args(
+    filepath_or_buffer,
+    object lineterminator,
+    object quotechar,
+    int quoting,
+    bool doublequote,
+    header,
+    bool mangle_dupe_cols,
+    usecols,
+    object sep,
+    object delimiter,
+    bool delim_whitespace,
+    bool skipinitialspace,
+    names,
+    dtype,
+    int skipfooter,
+    int skiprows,
+    bool dayfirst,
+    compression,
+    thousands,
+    decimal,
+    true_values,
+    false_values,
+    nrows,
+    byte_range,
+    bool skip_blank_lines,
+    parse_dates,
+    comment,
+    na_values,
+    bool keep_default_na,
+    bool na_filter,
+    object prefix,
+    index_col,
+) except +:
+
+    if delim_whitespace:
+        if delimiter is not None:
+            raise ValueError("cannot set both delimiter and delim_whitespace")
+        if sep != ',':
+            raise ValueError("cannot set both sep and delim_whitespace")
+
+    # Alias sep -> delimiter.
+    if delimiter is None:
+        delimiter = sep
+
+    if decimal == delimiter:
+        raise ValueError("decimal cannot be the same as delimiter")
+
+    if thousands == delimiter:
+        raise ValueError("thousands cannot be the same as delimiter")
+
+    if nrows is not None and skipfooter != 0:
+        raise ValueError("cannot use both nrows and skipfooter parameters")
+
+    if byte_range is not None:
+        if skipfooter != 0 or skiprows != 0 or nrows is not None:
+            raise ValueError("""cannot manually limit rows to be read when
+                                using the byte range parameter""")
+
+    cdef source_info c_source_info = make_source_info(filepath_or_buffer)
+    cdef read_csv_args read_csv_args_c = read_csv_args(c_source_info)
+
+    # Reader settings
+    compression = 'none' if compression is None else compression
+    compression = Compression[compression.upper()]
+    read_csv_args_c.compression = <compression_type> (
+        <underlying_type_t_compression> compression
+    )
+
+    if names is not None:
+        # explicitly mentioned name, so don't check header
+        read_csv_args_c.header = -1
+        read_csv_args_c.names.reserve(len(names))
+        for name in names:
+            read_csv_args_c.names.push_back(str(name).encode())
+    else:
+        if header is -1:
+            header_infer = 0
+        if header is None:
+            header_infer = -1
+
+    if prefix is not None:
+        read_csv_args_c.prefix = prefix.encode()
+
+    read_csv_args_c.mangle_dupe_cols = mangle_dupe_cols
+    read_csv_args_c.byte_range_offset = (
+        byte_range[0] if byte_range is not None else 0
+    )
+    read_csv_args_c.byte_range_size = (
+        byte_range[1] if byte_range is not None else 0
+    )
+
+    # Filter settings
+    if usecols is not None:
+        all_int = all([isinstance(col, int) for col in usecols])
```

Comment:
Use a generator instead of a list here as it will allow it to short circuit earlier if needed:
```suggestion
        all_int = all(isinstance(col, int) for col in usecols)
```